tools/autograd/gen_variable_type.py

```diff
@@ -181,6 +181,9 @@
 """)
 
 ADD_TRACE_INPUT = CodeTemplate("""jit::tracer::addInputs(node, "${name}", ${input});""")
+ADD_TRACE_INPUT_ALLOW_UNDEFINED = CodeTemplate("""\
+jit::tracer::addInputs(node, "${name}", ${input}, ${allow_undefined});
```

Comment:
nit: allow_undefined is only ever true in this template, so you can probably just hard code it.